game-core/src/main/java/games/strategy/engine/data/DelegateList.java

```diff
@@ -33,11 +33,11 @@ public IDelegate getDelegate(final String name) {
     return m_delegates.get(name);
   }
 
-  private void writeObject(final ObjectOutputStream out) {
+  private void writeObject(@SuppressWarnings("unused") final ObjectOutputStream out) {
     // dont write since delegates should be handled seperatly.
   }
 
-  private void readObject(final ObjectInputStream in) {
+  private void readObject(@SuppressWarnings("unused") final ObjectInputStream in) {
     m_delegates = new HashMap<>();
   }
```

Comment:
This looks like a smell to me.
If I understand this correctly, this object is not actually serializable, just behaves like it, and reinitialising this single field

Reply:
Yeah, this is an artifact of how the `GameDataManager#loadGame()` and `saveGame()` methods work.  A save game stream essentially consists of a `GameData` followed by zero or more delegates.  I believe the split was done to support cloning a `GameData` instance because you don't want the delegates in the clone to fire whenever the cloned `GameData` changes.

Since `GameDataManager` doesn't actually write the `DelegateList` itself, you're right that `DelegateList` does not need to be serializable.  I suspect, in the past, the delegates were always serialized with the `DelegateList` as part of the `GameData`, and when the clone optimization was introduced, they changed the implementation as above to retain compatibility.

Upon the next incompatible release, we probably should remove `Serializable` from this class and mark all such fields as transient.  I'll add a comment to do so as part of this PR.